internal/controller/runtime/fsm/runtime_fsm_create_shoot.go

```diff
@@ -15,6 +15,32 @@ const msgFailedToConfigureAuditlogs = "Failed to configure audit logs"
 func sFnCreateShoot(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
 	m.log.Info("Create shoot state")
 
+	if s.instance.Spec.Shoot.EnforceSeedLocation != nil && *s.instance.Spec.Shoot.EnforceSeedLocation {
+		seedAvailable, err := seedForRegionAvailable(m.ShootClient, s.instance.Spec.Shoot.Region)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to verify whether seed for region %s exists", s.instance.Spec.Shoot.Region)
+			m.log.Error(err, msg)
+			s.instance.UpdateStatePending(
+				imv1.ConditionTypeRuntimeProvisioned,
+				imv1.ConditionReasonGardenerError,
+				"False",
+				msg,
+			)
+			return updateStatusAndRequeueAfter(m.GardenerRequeueDuration)
+		}
+
+		if !seedAvailable {
+			msg := fmt.Sprintf("Seed for region %s doesn't exist", s.instance.Spec.Shoot.Region)
```

Comment:
Align with KEB is:

* We provide in the error a list of available regions which include a seed
* We use a dedicated Error-Type for Seed-Not-Found reason

Reply:
Long term solution is to reply an JSON string in the response to be able to provide more meta-data to KEB.